src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2895,20 +2893,15 @@ void func_80835F44(GlobalContext* globalCtx, Player* this, s32 item) {
                 } else {
                     func_80078884(NA_SE_SY_ERROR);
                 }
-                return;
-            }
 
-            if (actionParam == PLAYER_AP_NUT) {
+            } else if (actionParam == PLAYER_AP_NUT) {
                 if (AMMO(ITEM_NUT) != 0) {
                     func_8083C61C(globalCtx, this);
                 } else {
                     func_80078884(NA_SE_SY_ERROR);
                 }
-                return;
-            }
 
-            temp = Player_ActionToMagicSpell(this, actionParam);
-            if (temp >= 0) {
+            } else if ((temp = Player_ActionToMagicSpell(this, actionParam)) >= 0) {
```

Comment:
yeah I figured this inline assignment had something to do with the early returns, but this chain of early returns really sticks out with this chain of ifs

what also made me decide this inline assignment is "fine" is it's already done above in the same function with `temp = Player_ActionToExplosive(this, actionParam),` 😅 so it really feels like a pattern

Reply:
I need some time to think this one over. Im super familiar with this function cause ive had to read many times and modify it for other projects. Keeping this function as readable as possible is super important for anyone who wants to modify player

Currently i say early returns probably make this easier to read and understand, but want to see how i feel about it after letting it sit some more

Reply:
To me early returns make it less easy to understand/modify the code because the control flow is less clear than with `else`

Options:

1) current in this PR
```c
} else if ((temp = Player_ActionToMagicSpell(this, actionParam)) >= 0) {
```

2) assignment in the if but "not inline"
```c
} else if (temp = Player_ActionToMagicSpell(this, actionParam), temp >= 0) {
```

3) a whole new else block
```c
            } else {
                temp = Player_ActionToMagicSpell(this, actionParam);
                if (temp >= 0) {
                    if (...
                // everything else with one more indentation level than needed
```

4) early returns before and keep the last `else`s
```c
            if ((globalCtx->bombchuBowlingStatus == 0) &&
                (((actionParam == PLAYER_AP_STICK) && (AMMO(ITEM_STICK) == 0)) ||
                 ((actionParam == PLAYER_AP_BEAN) && (AMMO(ITEM_BEAN) == 0)) ||
                 (temp = Player_ActionToExplosive(this, actionParam),
                  ((temp >= 0) && ((AMMO(sExplosiveInfos[temp].itemId) == 0) ||
                                   (globalCtx->actorCtx.actorLists[ACTORCAT_EXPLOSIVE].length >= 3)))))) {
                func_80078884(NA_SE_SY_ERROR);

                return;
            } else if (actionParam == PLAYER_AP_LENS) {
                if (func_80087708(globalCtx, 0, 3)) {
                    if (globalCtx->actorCtx.lensActive) {
                        Actor_DisableLens(globalCtx);
                    } else {
                        globalCtx->actorCtx.lensActive = true;
                    }
                    func_80078884((globalCtx->actorCtx.lensActive) ? NA_SE_SY_GLASSMODE_ON : NA_SE_SY_GLASSMODE_OFF);
                } else {
                    func_80078884(NA_SE_SY_ERROR);
                }

                return;
            } else if (actionParam == PLAYER_AP_NUT) {
                if (AMMO(ITEM_NUT) != 0) {
                    func_8083C61C(globalCtx, this);
                } else {
                    func_80078884(NA_SE_SY_ERROR);
                }

                return;
            }

            temp = Player_ActionToMagicSpell(this, actionParam);
            if (temp >= 0) {
```

Reply:
after checking this out and reading through it in my IDE, it looks *alot* better as is. The temp being assigned in the if sucks, but it seems worth it for the other improvements it allows.

I would handle line breaks a bit differently here, I sent a pr with what my preference would be. Was too much to mark up in a review

Reply:
```c
} else if ((temp = Player_ActionToMagicSpell(this, actionParam)) >= 0) {
```
can also still be changed to
```c
} else if (temp = Player_ActionToMagicSpell(this, actionParam), temp >= 0) {
```
if that would be any better

Reply:
the extra stuff with the comma makes it more confusing I think, but can hear other opinions